src/backlight/portfolio/portfolio.py

```diff
@@ -0,0 +1,250 @@
+import pandas as pd
+import numpy as np
+from typing import List
+
+from backlight.positions.positions import Positions
+from backlight.datasource.marketdata import MarketData
+from backlight.metrics.position_metrics import calc_pl
+from backlight.positions import calc_positions
+from backlight.trades.trades import Trades
+from joblib import Parallel, delayed
+
+
+class Portfolio:
+    """
+    An abstract definition of a Portfolio is defined as a list of Positions
+    Each element of the list represent the positions of an asset that contribute to the portfolio
+    """
+
+    def __init__(self, positions: List[Positions]):
+        """
+        If there is positions with the same symbol, their value are sum
+        """
+        symbols = [position.symbol for position in positions]
+        if len(symbols) > len(set(symbols)):
```

Comment:
You can just `assert len(symbols) == len(set(symbols))`.
`Portfolio` class just assume that the inputed `positions: List[Positions]`' symbol are unique.

Below lines should be moved to `construct_portfolio` later.